### PR TITLE
[DOXIA-685] Replace SinkEventAttributes#BOXED and clearly separate between regular verbatim and verbatim source (code)

### PR DESCRIPTION
**Pull request by michael-o**

### CODE REVIEW

**Comment by michael-o** [comment]
@hboutemy Please review on the weekend. This is the last change I want to merge for M5 and perform a release.

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java [hunk 1]
```diff
@@ -115,9 +115,9 @@ public final void testTestDocument() throws IOException {
     }
 
     /**
-     * Checks that the sequence <code>[title(), text( title ), title_()]</code>,
+     * Checks that the sequence <code>[title(), text(title), title_()]</code>,
```
Comment (hboutemy) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:118]: these javadoc formatting updates add noise to the change: can we split them in a different PR (no need for any Jira issue...)

**Review by hboutemy — CHANGES_REQUESTED** [review]
please extract javadoc formatting updates (I'll try to do it you you want)

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java (hunk 1 above)
Comment (michael-o) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:118]: Had the same concern. Will move them out today.

Comment (hboutemy) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java]: I just did it and rebased your branch

**Review by hboutemy — APPROVED** [review]
split between javadoc and interesting code change done

**Comment by jira-importer** [comment]
Resolve #830
